### PR TITLE
Move phone number verification to identity

### PR DESCRIPTION
**Pull request by lazynina**
Screenshot from 2021-11-12 12-56-30

Screenshot from 2021-11-12 13-00-04

image

### CODE REVIEW

File: src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.ts [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+          console.error(err);
+        }).add(() => this.loading = false);
+      } else {
+        this.loading = false;
+      }
+    });
+
+  }
+
+  _setScreenToShow() {
```
Comment (maebeam) [inline src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.ts:74]: Why is this necessary? Why not just put this in the GetUsersStateless callback?

**Review by maebeam — COMMENTED** [review]
LGTM, would love to remove the interval if possible

File: src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.ts (hunk 1 above)
Comment (lazynina) [reply src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.ts:74]: Yeah I should move this. I just copied this from the code that was in frontend.

File: src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.html [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+          *ngIf="displayForSignupFlow"
+          (click)="onSkipButtonClicked()"
+          class="btn btn-outline-primary font-weight-bold fs-15px"
+        >
+          Skip
+        </a>
+        <a
+          (click)="sendVerificationText()"
+          [ngClass]="{ disabled: !phoneForm.valid, 'btn-loading': sendingPhoneNumberVerificationText }"
+          class="btn btn-primary font-weight-bold fs-15px ml-10px"
```
Comment (superzordon) [inline src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.html:105]: Use classes `button button-primary button-small` instead of `btn btn-primary` to match new identity theme

File: src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.html [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        </div>
+      </div>
+    </form>
+
+    <div class="d-flex mt-15px justify-content-between">
+      <div>
+        <a
+          *ngIf="displayForSignupFlow"
+          (click)="backButtonClickedOnSubmitVerificationScreen()"
+          class="btn btn-outline-primary font-weight-bold fs-15px"
```
Comment (superzordon) [inline src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.html:152]: `button button-secondary button-small`

File: src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.html [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        >
+          Back
+        </a>
+      </div>
+      <div>
+        <a
+          *ngIf="displayForSignupFlow"
+          (click)="onSkipButtonClicked()"
+          class="btn btn-outline-primary font-weight-bold fs-15px"
```
Comment (superzordon) [inline src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.html:161]: `button button-secondary button-small`

File: src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.html [hunk 5]
```diff
@@ -0,0 +1,182 @@
+<app-banner></app-banner>
+<ng-template #continueToHost>
+  <div class="d-flex pt-15px justify-content-between">
+    <button
+      (click)="finishFlow()"
+      class="btn btn-outline-primary font-weight-bold fs-15px sign-up-btn"
```
Comment (superzordon) [inline src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.html:6]: `button button-primary button-large`

File: src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.html [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    <div class="fs-10px">
+      By proceeding, you agree to receive text messages from deso.org (standard rates apply)
+    </div>
+
+    <div class="d-flex mt-15px justify-content-end">
+      <div>
+        <a
+          *ngIf="displayForSignupFlow"
+          (click)="onSkipButtonClicked()"
+          class="btn btn-outline-primary font-weight-bold fs-15px"
```
Comment (superzordon) [inline src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.html:98]: `button button-secondary button-small`

File: src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.html [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+        >
+          Skip
+        </a>
+        <a
+          (click)="submitVerificationCode()"
+          [ngClass]="{ disabled: !verificationCodeForm.valid, 'btn-loading': submittingPhoneNumberVerificationCode }"
+          class="btn btn-primary font-weight-bold fs-15px ml-10px"
```
Comment (superzordon) [inline src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.html:168]: `button button-primary button-small`

**Review by superzordon — COMMENTED** [review]
Just a couple of minor formatting nitpicks to keep this page consistent with the rest of identity.

File: src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.ts [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+    }
+
+    // Clear any existing resend-related errors
+    this.sendPhoneNumberVerificationTextServerErrors = new SendPhoneNumberVerificationTextServerErrors();
+
+    this._sendPhoneNumberVerificationText();
+
+    // Display a success indicator
+    this.resentVerificationCode = true;
+    setTimeout(() => (this.resentVerificationCode = false), 5000);
```
Comment (AeonSw4n) [inline src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.ts:124]: This is working, right? Was just thinking if we might need a `.bind(this)` here, but it's probably fine... `this` scopes always confuse me.

**Review by AeonSw4n — APPROVED** [review]
LGTM! The main verification part seems safe. Also, I didn't check the backend endpoints, but I'm assuming everything's chill in there. :+1:

File: src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.ts (hunk 8 above)
Comment (diamondhands0) [reply src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.ts:124]: This thing `()=>{}` keeps `this` as whatever it was in the parent scope. It's generally always what you want.

**Review by diamondhands0 — APPROVED** [review]
